test/fuzz/lua/test_mvcc.lua

```diff
@@ -0,0 +1,845 @@
+    local rw_txs = {}
+    local ro_txs = {}
+
+    local tx_operations = {}
+    for _ = 1, TX_CNT do
+        table.insert(tx_operations, {})
+    end
+
+    for i = 1, #stmts do
+        local stmt = stmts[i]
+        if not committed_txs_mask[stmt.tid] then
+            goto continue
+        end
+
+        if stmt.str:find('begin') and committed_txs_mask[stmt.tid] and
+           ro_txs_mask[stmt.tid] and not bad_dml_txs_mask[stmt.tid] then
+            table.insert(ro_txs, stmt.tid)
+        end
+
+        if stmt.str:find('commit') and committed_txs_mask[stmt.tid] and
+           not ro_txs_mask[stmt.tid] then
+            table.insert(rw_txs, stmt.tid)
+        end
+
+        if stmt.type ~= TCL then
+            table.insert(tx_operations[stmt.tid], stmt)
+        end
+        ::continue::
+    end
+
+    -- First of all, try to serialize read-only transactions.
+    for i, tid in ipairs(ro_txs) do
+        if try_apply_tx(tx_operations[tid], false) then ro_txs[i] = nil end
+    end
+
+    for _, rw_tid in ipairs(rw_txs) do
+        -- Try to serialize read-write transactions.
+        try_apply_tx(tx_operations[rw_tid], true)
+        -- Afterwards, try to serialize read-only transactions again.
+        for i, ro_tid in pairs(ro_txs) do
+            if try_apply_tx(tx_operations[ro_tid], false) then
+                ro_txs[i] = nil
+            end
+        end
+    end
+
+    if next(ro_txs) ~= nil then
+        local failed_ro_set = {}
+        for _, tid in pairs(ro_txs) do
+            table.insert(failed_ro_set, tid)
+        end
+        dump_repro()
+        dump_serial()
+        local msg = 'failed to serialize the following read-only ' ..
+            'transactions: ' .. table.concat(failed_ro_set, ', ')
+        log.info(msg)
+        os.exit(1)
+    end
+end
+
+local function open_files()
+    local repro_file_name = 'repro.lua'
+    local repro_file_path = fio.pathjoin(mvcc.workdir, '..',
+                                         repro_file_name)
+    local err
+    repro_file, err = fio.open(repro_file_path,
+                               {'O_WRONLY', 'O_CREAT', 'O_TRUNC'},
+                               {'S_IRUSR', 'S_IWUSR'})
+    if repro_file == nil then
+        log.info('fio.open failed: ' .. err)
+        os.exit(1)
+    end
+
+    local serial_file_name = 'serial.lua'
+    local serial_file_path = fio.pathjoin(oracle.workdir, '..',
+        serial_file_name)
+    serial_file, err = fio.open(serial_file_path,
+                               {'O_WRONLY', 'O_CREAT', 'O_TRUNC'},
+                               {'S_IRUSR', 'S_IWUSR'})
+    if serial_file == nil then
+        log.info('fio.open failed: ' .. err)
+        os.exit(1)
+    end
+end
+
+local function close_files()
+    repro_file:close()
+    serial_file:close()
+end
+
+local function setup_test()
+    math.randomseed(SEED)
+    log.info("Random seed: %d", SEED)
+
+    oracle = server:new{alias = 'oracle'}
+    mvcc = server:new{
```

Comment:
call with round brackets

Reply:
>  Even our documentation has examples with calls with round brackets

I suppose these examples were added many times ago due to this use of old style.
Most Lua code in Tarantool uses round brackets.